src/backend/distributed/commands/dependencies.c

```diff
@@ -224,15 +226,37 @@ GetDependencyCreateDDLCommands(const ObjectAddress *dependency)
 	{
 		case OCLASS_CLASS:
 		{
+			char relKind = get_rel_relkind(dependency->objectId);
+
 			/*
 			 * types have an intermediate dependency on a relation (aka class), so we do
 			 * support classes when the relkind is composite
 			 */
-			if (get_rel_relkind(dependency->objectId) == RELKIND_COMPOSITE_TYPE)
+			if (relKind == RELKIND_COMPOSITE_TYPE)
 			{
 				return NIL;
 			}
 
+			if (relKind == RELKIND_RELATION || relKind == RELKIND_PARTITIONED_TABLE ||
+				relKind == RELKIND_FOREIGN_TABLE)
+			{
+				Oid relationId = dependency->objectId;
+				List *commandList = NIL;
+
+				if (IsCitusTable(relationId) && !IsTableOwnedByExtension(relationId))
```

Comment:
I think it is too late to call `IsTableOwnedByExtension ` at this point. We should not have gotten here for objects owned by extensions.

For example, we guard against  objects owned by extension in `ApplyAddToDependencyList` via `IsObjectAddressOwnedByExtension`.

And, in `ReplicateAllDependenciesToNodeCommandList` we call `ApplyAddToDependencyList` via `OrderObjectAddressListInDependencyOrder`.

So, I think we should remove this from there.

And, I think it is valuable to add a regression test such that we add new node when we have tables owned by extensions.  Can you please add such test in this PR?

An example of tables owned by extension: https://github.com/citusdata/citus/pull/3938/files#diff-cb18af43e1dbb9f6dd703587058353f0e4e9fabd881216a3da036bcb3e31990eR137